src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala

```diff
@@ -198,7 +198,7 @@ class JdbcReadJournal(config: Config)(implicit val system: ExtendedActorSystem)
                   if (hasMoreEvents) Continue else ContinueDelayed
               }
 
-            val nextStartingOffset = if (xs.isEmpty) from else xs.map(_.offset.value).max
+            val nextStartingOffset = if (xs.isEmpty) math.max(from, queryUntil.maxOrdering) else xs.map(_.offset.value).max
```

Comment:
maybe we should add a comment to explain why this is needed and which special case we are covering. 

I remember vaguely that we have discussed it, but it's now somewhere deep in my memory.